test/git-dir-handler.ts

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, before, afterEach} from 'mocha';
+import {assert, expect} from 'chai';
+import {setup} from './util';
+import {
+  getGitFileData,
+  getAllDiffs,
+  parseChanges,
+  findRepoRoot,
+  resolvePath,
+} from '../src/bin/handle-git-dir-change';
+import * as fs from 'fs';
+import * as sinon from 'sinon';
+import * as child_process from 'child_process';
+
+before(() => {
+  setup();
+});
+
+// tslint:disable:no-unused-expression
+// .null triggers ts-lint failure, but is valid chai
+// .true triggers ts-lint failure, but is valid chai
+describe('git directory diff output to git file data + content', () => {
```

Comment:
nit: usually the describe block shows the function name under test
```suggestion
describe('getGitFileData', () => {
```